### PR TITLE
Java Console reconnection fixing (by issue #3305)

### PR DESCRIPTION
**Pull request by alxrMironovHome**
With this changes JC is reconnecting after the 5s timeout (as planned???)

### CODE REVIEW

File: java_console/io/src/main/java/com/rusefi/io/LinkManager.java [hunk 1]
```diff
@@ -46,6 +46,9 @@ public String unpack(String packedLine) {
     public static final String LOG_VIEWER = "log viewer";
     private final CommandQueue commandQueue;
 
+
+    private String lastTriedPort;
```
Comment (mck1117) [inline java_console/io/src/main/java/com/rusefi/io/LinkManager.java:50]: please explicitly initialize (perhaps to null)

File: java_console/io/src/main/java/com/rusefi/io/LinkManager.java [hunk 2]
```diff
@@ -46,8 +46,10 @@ public String unpack(String packedLine) {
     public static final String LOG_VIEWER = "log viewer";
     private final CommandQueue commandQueue;
 
+    private String lastTriedPort = null;
+
     private LinkConnector connector = LinkConnector.VOID;
-    private boolean isStarted;
+    private boolean isStarted = false;
```
Comment (rusefillc) [inline java_console/io/src/main/java/com/rusefi/io/LinkManager.java:52]: no reason to change that line

File: java_console/io/src/main/java/com/rusefi/io/LinkManager.java [hunk 3]
```diff
@@ -272,13 +275,22 @@ public void send(String command, boolean fireEvent) throws InterruptedException
...
-        connector.restart();
+        close(); // Explicitly kill the connection (call connectors destructor??????)
+
+        String[] ports = getCommPorts();
+        boolean isPortAvaliableAgain = Arrays.stream(ports).anyMatch(lastTriedPort::equals);
+        if(isPortAvaliableAgain) {
+            connect(lastTriedPort);
+        }
+
+       // connector.restart();
```
Comment (rusefillc) [inline java_console/io/src/main/java/com/rusefi/io/LinkManager.java:286]: we do not need dead code, please remove if if you are removing it

**Review by rusefillc — CHANGES_REQUESTED** [review]
this removes LinkConnector#restart usages so let's remove all LinkConnector#restart implementations and code

overall my problem is that I do not understand this change. what is fundamentally different between new code and old code?

**Pull request merged**